sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -943,4 +944,52 @@ public void test_update_by_id_where_no_row_is_matching() throws Exception {
         assertThat(response.cols()[0], is("id"));
         assertThat(response.rowCount(), is(0L));
     }
+
+    @Test
+    public void test_update_by_query_returning_single_field_with_outputputname() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where message='msg' returning message as message_renamed");
+
+        assertThat((response.rowCount()), is(1L));
+        assertThat((response.cols()[0]), is("message_renamed"));
+        assertThat(response.rows()[0][0], is("updated"));
+    }
+
+
+    @Test
+    public void test_update_by_query_with_subquery_returning_multiple_fields() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where message= (select 'msg') returning id, message");
+
+        assertThat((response.rowCount()), is(1L));
+        assertThat((response.cols()[0]), is("id"));
+        assertThat((response.cols()[1]), is("message"));
+        assertThat(response.rows()[0][0], is(1));
+        assertThat(response.rows()[0][1], is("updated"));
+
+    }
+
+    @Test
+    public void test_update_by_query_returning_multiple_results() throws Exception {
+        execute("create table test (id int primary key, x int, message string) clustered into 2 shards");
+        execute("insert into test values(1, 1, 'msg');");
+        execute("insert into test values(2, 1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where message='msg' and x > 0 returning message as message_renamed");
```

Comment:
Maybe we could also add `_seq_no` or so to the `RETURNING` clause here. As that is one of the use cases we want to enable